app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,17 @@
+class CommentsController < ApplicationController
+  before_action :authenticate_user!
+
+  def create
+    @comment = current_user.comments.build(comment_params)
```

Comment:
Since you nested the controller with the books one (which is good) you could create a `helper_method` (http://apidock.com/rails/ActionController/Helpers/ClassMethods/helper_method) to find that book and later define it as a private method like this:
```ruby
helper_method :book

private 

def book
  @book ||= Book.find(params[:book_id]
end
```

Finally you can change the controller method as following:
```ruby
def create
  @comment = book.comments.create(comments_params)
  redirect_to :back
end
```